lib/Internal/Placeholder.php

```diff
@@ -84,15 +84,18 @@ private function resolve($value = null) {
                 $trace = formatStacktrace($this->resolutionTrace);
                 $message .= ". Previous resolution trace:\n\n{$trace}\n\n";
             } else {
-                $message .= ", define const AMP_DEBUG = true and enable assertions for a stacktrace of the previous resolution.";
+                // @codeCoverageIgnoreStart
+                $message .= ", define environment variable AMP_DEBUG or const AMP_DEBUG = true and enable assertions "
+                    . "for a stacktrace of the previous resolution.";
+                // @codeCoverageIgnoreEnd
             }
 
             throw new \Error($message);
         }
 
         \assert((function () {
-            if (\defined("AMP_DEBUG") && \AMP_DEBUG) {
-                $trace = \debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
+            if (\getenv("AMP_DEBUG") || (\defined("AMP_DEBUG") && \AMP_DEBUG)) {
```

Comment:
Should the env also be checked for a truethy value?

Reply:
It is, but I think I should provide a check against `"false"` if that's what you mean.